lib/tasks/data.rake

```diff
@@ -39,64 +39,77 @@ namespace :data do
+        if history_tasks.size > 0
+          task1 = history_tasks.last
+          task2 = history_tasks.select {|task| task.host != task1.host}.last
 
-            if Socket.gethostname == task1.host
-              # last time it was run on task1.host. Yield to the other instance this time.
-              puts "[#{Time.now}] Cron job #{task} has been run on #{task1.host}. Stop."
-              return
-            elsif task2.present? && Socket.gethostname == task2.host # && Socket.gethostname != task1.host
-              # last time it was run on task2.host. Yield to the other instance this time.
-              puts "[#{Time.now}] Cron job #{task} has been run on #{task2.host}. Stop."
-              return
-            elsif task2.present? # && Socket.gethostname != task1.host && Socket.gethostname != task2.host
-              # Both task1 and task2 exist in history
-              # However, IPs of both the instances have changed (after a new deployment).
-              puts "[#{Time.now}] This is possibly a new deployment. Wait randomly for 0 to 60 seconds and retry."
-              sleep rand(0..60)
-              tried += 1
-              puts "[#{Time.now}] Wait is done. Retrying (#{tried}) on #{Socket.gethostname}."
-              if tried == 10
-                puts "[#{Time.now}] Cron job #{task} is being started on #{Socket.gethostname}."
-                job = CronJobHistory.new(task_name: task, last_run: Time.now, status: 'running', host: Socket.gethostname)
-                job.save!
-                id = job.id
-                yield
-              else
-                next
-              end
-            else # task2.nil?
-              # task1 not nil. But the host name is different now which means it is no different than a fresh deployment.
-              puts "[#{Time.now}] Cron job #{task} has been run for #{history_tasks.size} times (in total from both hosts) during the past #{(1.5 * interval).to_i / 3600.0} hours."
+          if Socket.gethostname == task1.host
+            # last time it was run on task1.host. Yield to the other instance this time.
+            puts "[#{Time.now}] Cron job #{task} has been run on #{task1.host}. Stop."
+            return
+          elsif task2.present? && Socket.gethostname == task2.host # && Socket.gethostname != task1.host
+            # last time it was run on task2.host. Yield to the other instance this time.
+            puts "[#{Time.now}] Cron job #{task} has been run on #{task2.host}. Stop."
+            return
+          elsif task2.present? # && Socket.gethostname != task1.host && Socket.gethostname != task2.host
+            # Both task1 and task2 exist in history
+            # However, IPs of both the instances have changed (after a new deployment).
+            puts "[#{Time.now}] This is possibly a new deployment. Wait randomly for 0 to 60 seconds and retry task #{task}."
+            sleep rand(0..60)
```

Comment:
I realize this is likely fine as is (it was in the old code even and so apparently was working) - but waiting a 'random' amount of time seems inconsistent to me.  Should we consider changing this to something consistent?  Would it be terrible if it waited a full minute each time?  Just a thought.